src/Control/Carrier/Simple.hs

```diff
@@ -120,46 +124,52 @@ import Control.Applicative
 import Control.Carrier.Reader
 import Control.Carrier.State.Strict
 import Control.Monad.Trans
-import Data.Kind (Type)
-import Unsafe.Coerce (unsafeCoerce)
+
+---------- The Simple effect
+
+-- | Simple effects.. as an effect
+data Simple e m a where
```

Comment:
[no action] Why simple as name (can't imagine what would be opposite of it 😅)- Is this conventional?

Reply:
the real term is "first-order" effects (as opposed to "higher-order").. could definitely rename and add docs for what it means. would probably be useful

Reply:
the tl;dr is first-order effects don't contain effectful sub-expressions:

example first-order:
```hs
data Filesystem m a where
  ReadFile :: FilePath -> Filesystem m String
```

example higher-order:
```hs
data Error e m a where
  Catch :: m a -> (e -> m a) -> Error e m a

catch :: Has (Error e) sig m => m a -> (e -> m a) -> m a
catch act hdl = send (Catch act hdl)

-- or from our codebase
data Diagnostics m a where
  Context :: String -> m a -> Diagnostics m a

context :: Has Diagnostics sig m => String -> m a -> m a
context ctx act = send (Context ctx act)
```

"higher-order" effects are ones where `m` appears as an argument to an effect constructor.

`Simple` is a wrapper effect around first-order effects that doesn't even allow the effect type to mention `m`. e.g., in this PR:

```hs
data ReadFSF a where
  DoesDirExist :: ... -> ReadFSF Bool

type ReadFS = Simple ReadFSF

doesDirExist :: Has ReadFS sig m => ... -> m Bool
doesDirExist dir = send (Simple (DoesDirExist dir))
```